akd/src/directory.rs

```diff
@@ -872,3 +872,196 @@ pub async fn get_directory_root_hash_and_ep<
     let root_hash = akd_dir.get_root_hash::<H>(&current_azks).await?;
     Ok((root_hash, latest_epoch))
 }
+
+/// Helpers for testing
+
+#[cfg(test)]
+#[derive(Debug, Clone)]
+pub(crate) enum PublishCorruption {
+    /// Indicates to the malicious publish function to not mark a stale version
+    UnmarkedStaleVersion(AkdLabel),
+    /// Indicates to the malicious publish to mark a certain version for a username as stale.
+    MarkVersionStale(AkdLabel, u64),
+    /// No corruption
+    NoCorruption,
```

Comment:
@Jasleen1 : Seems like NoCorruption isn't used anywhere. Can we delete this to resolve the warning?

Also, can we move this code into tests.rs?

Reply:
Discussed offline with Jasleen, it should be ok to keep in directory.rs. Let's land this!